tests/test_basic_auth.py

```diff
@@ -124,6 +124,21 @@ def test_auth_token_precedence():
         "With multiple authentication options, auth_token string should take highest precendence"
 
 
+@pytest.mark.usefixtures("mock_domino_version_response")
+@pytest.mark.skipif(not os.getenv(DOMINO_TOKEN_FILE_KEY_NAME), reason="No token file in environment")  # noqa:E501
+def test_auth_with_only_api_key_passed_as_parameter():
+    """
+    Confirm that api key takes precedence over both token file and API key authentication
+    if they are not passed as parameters.
+    """
+    dummy_host = "http://domino.somefakecompany.com"
+    dummy_api_key = "top_secret_api_key"
+
+    d = Domino(host=dummy_host, project="anyuser/quick-start", api_key=dummy_api_key)
+    assert isinstance(d.request_manager.auth, requests.auth.HTTPBasicAuth), \
+        "With only api key passed as a parameter, it takes precedence over DOMINO_TOKEN_FILE_KEY_NAME"
```

Comment:
This is an exact duplicate of `test_object_creation_with_api_key` above.
```suggestion
```

Reply:
If you really want to be thorough, these are tests that you could consider adding.

- set a value for `DOMINO_USER_API_KEY_KEY_NAME` in the environment, also pass a different API key as a parameter, and confirm that the parameter is used
- set keys in the os.environ for both `DOMINO_TOKEN_FILE_KEY_NAME` and `DOMINO_USER_API_KEY_KEY_NAME`, and confirm that `DOMINO_TOKEN_FILE_KEY_NAME` gets used when no explicit auth is passed
- set a key in the os.environ for `DOMINO_USER_API_KEY_KEY_NAME` only, and confirm that it gets used when no other explicit auth is passed

Reply:
> This is an exact duplicate of `test_object_creation_with_api_key` above.

It isn't. The new test assumes there is a token file found in the env, that's the test actually testing the changes :) Let me rename the test method to make it clearer.